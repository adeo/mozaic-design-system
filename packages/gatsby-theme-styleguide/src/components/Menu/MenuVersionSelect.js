import React, { Component } from 'react'
import styled from 'styled-components'

const SelectContainer = styled.div`
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;

  select {
    display: block;
    cursor: pointer;
    height: 100%;
    opacity: 0;
    width: 100%;
    z-index: 2;
  }
`

const LabelVersion = styled.label`
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  white-space: nowrap;
  border: 0;
`

const MenuVersionRelease = ({ releases }) => {
  const handleChange = (event) => {
    if (event.target.value !== 'default')
      window.location.href = event.target.value
  }

  return (
    <SelectContainer>
      <LabelVersion>older tags:</LabelVersion>
      <select defaultValue="default" onChange={handleChange}>
        <option key="default" value="default">
          -- Choose version --
        </option>
        <option key="latest" value="https://mozaic.adeo.cloud/">
          latest release
        </option>
        {releases.map((release) => (
          <option key={release.tagName} value={release.url}>
            {release.tagName}
          </option>
        ))}
      </select>
    </SelectContainer>
  )
}

export default MenuVersionRelease
