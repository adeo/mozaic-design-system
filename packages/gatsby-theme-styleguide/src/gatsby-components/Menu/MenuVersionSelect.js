import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const SelectContainer = styled.div`
  background: #222020;
  bottom: 0;
  left: 0;
  padding: 2rem 1.5rem;
  position: absolute;
  width: 100%;

  select {
    display: block;
    border: none;
    border-radius: 0.25rem;
    background: #fff;
    width: 100%;
    font-size: 100%;
    padding: 0.312rem 1.125rem 0.375rem 0.25rem;
  }
`

const LabelVersion = styled.label`
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  white-space: nowrap;
  border: 0;
`

class MenuVersionRelease extends Component {
  handleChange = event => {
    if (event.target.value !== 'default')
      window.location.href = event.target.value
  }

  static propTypes = {
    githubReleases: PropTypes.arrayOf(
      PropTypes.shape({
        node: PropTypes.shape({
          url: PropTypes.string.isRequired,
          tagName: PropTypes.string.isRequired,
          isCurrent: PropTypes.bool.isRequired,
        }).isRequired,
      })
    ).isRequired,
  }

  render() {
    const githubReleases = [...this.props.githubReleases].reverse()

    return (
      <SelectContainer>
        <LabelVersion>older tags:</LabelVersion>
        <select defaultValue="default" onChange={this.handleChange}>
          <option key="default" value="default">
            -- Choose version --
          </option>
          <option key="latest" value="https://mozaic.adeo.cloud/">
            latest release
          </option>
          {githubReleases.reverse().map(release => (
            <option key={release.node.tagName} value={release.node.url}>
              {release.node.tagName}
            </option>
          ))}
        </select>
      </SelectContainer>
    )
  }
}

export default MenuVersionRelease
