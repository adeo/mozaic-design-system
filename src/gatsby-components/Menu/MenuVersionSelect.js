import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const LabelVersion = styled.label`
  font-size: 0.5rem;
  font-weight: normal;
  line-height: 1;
  letter-spacing: 0.1em;
  margin-right: 5px;
`

class MenuVersionRelease extends Component {
  handleChange = event => {
    window.location.href = event.target.value
  }

  static propTypes = {
    githubReleases: PropTypes.arrayOf(
      PropTypes.shape({
        node: PropTypes.shape({
          url: PropTypes.string.isRequired,
          tagName: PropTypes.string.isRequired,
          isCurrent: PropTypes.bool.isRequired,
        }).isRequired,
      })
    ).isRequired,
  }

  render() {
    const githubReleases = [...this.props.githubReleases].reverse()

    return (
      <div>
        <LabelVersion>older tags:</LabelVersion>
        <select onChange={this.handleChange}>
          {githubReleases.reverse().map(release => (
            <option
              key={release.node.tagName}
              value={release.node.url}
              selected={release.node.isCurrent}
            >
              {release.node.tagName}
            </option>
          ))}
        </select>
      </div>
    )
  }
}

export default MenuVersionRelease
