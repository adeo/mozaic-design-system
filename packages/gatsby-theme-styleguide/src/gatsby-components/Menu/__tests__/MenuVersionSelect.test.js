import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import MenuVersionSelect from '../MenuVersionSelect'

const props = {
  githubReleases: [
    {
      node: { url: 'http://1', tagName: '1', isCurrent: false },
    },
    {
      node: { url: 'http://2', tagName: '2', isCurrent: true },
    },
  ],
}

describe('<MenuVersionSelect />', () => {
  beforeEach(function() {
    global.window = Object.create(window)
    Object.defineProperty(window, 'location', {
      value: {
        href: 'http://dummy.com',
      },
      writable: true,
    })
  })

  it('render corectly', () => {
    const tree = renderer.create(<MenuVersionSelect {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('changes the url when we change the select option', () => {
    const wrapper = mount(<MenuVersionSelect {...props} />)
    wrapper.find('select').simulate('change', { target: { value: 'http://2' } })
    expect(window.location.href).toEqual('http://2')
  })

  it('does not change anything if we change select to default value', () => {
    const wrapper = mount(<MenuVersionSelect {...props} />)
    wrapper.find('select').simulate('change', { target: { value: 'default' } })
    expect(window.location.href).toEqual('http://dummy.com')
  })
})
