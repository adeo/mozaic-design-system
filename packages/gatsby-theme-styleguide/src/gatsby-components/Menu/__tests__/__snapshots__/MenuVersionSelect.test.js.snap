// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuVersionSelect /> render corectly 1`] = `
.c0 {
  background: #222020;
  bottom: 0;
  left: 0;
  padding: 2rem 1.5rem;
  position: absolute;
  width: 100%;
}

.c0 select {
  display: block;
  border: none;
  border-radius: 0.25rem;
  background: #fff;
  width: 100%;
  font-size: 100%;
  padding: 0.312rem 1.125rem 0.375rem 0.25rem;
}

.c1 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border: 0;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    older tags:
  </label>
  <select
    defaultValue="default"
    onChange={[Function]}
  >
    <option
      value="default"
    >
      -- Choose version --
    </option>
    <option
      value="https://mozaic.adeo.cloud/"
    >
      latest release
    </option>
    <option
      value="http://1"
    >
      1
    </option>
    <option
      value="http://2"
    >
      2
    </option>
  </select>
</div>
`;
